### ISSUE TITLE
MultiSelect Chips mode value template behavior

### ISSUE THREAD

**Issue by reijosirila** [primefaces/primevue#5926]
### Describe the bug

In PrimeVue 4, the MultiSelect component's behavior has changed such that using the "value" template overrides the entire chip instead of just the value inside the chip. Additionally, the "value" parameter now represents the entire array of selected values instead of individual values, which differs from the behavior in PrimeVue 3.

**Expected Behavior**
In PrimeVue 3, when using the MultiSelect component with chips and a "value" template, the chip should render with the value inside it being overridden according to the template. The "value" parameter should represent individual selected values.

**Actual Behavior**
In PrimeVue 4, using the "value" template on the MultiSelect component:

Overrides the entire chip instead of just the value inside the chip.
The "value" parameter represents the entire array of selected values rather than individual values.

### Reproducer

https://stackblitz.com/edit/primevue-nuxt-issue-template

### PrimeVue version

4.0.0-rc.2

### Vue version

3.x

### Language

TypeScript

### Build / Runtime

Vue CLI App

### Steps to reproduce the behavior

Create a Vue component with the PrimeVue MultiSelect.
Use the MultiSelect component with display="chip" and a custom "value" template.
Observe the rendered output and console.

Example code: 
```
<script setup lang="ts">
import { ref } from 'vue';

const selectedCountries = ref();
const countries = ref([
  { name: 'Australia', code: 'AU' },
  { name: 'Brazil', code: 'BR' },
  { name: 'China', code: 'CN' },
  { name: 'Egypt', code: 'EG' },
  { name: 'France', code: 'FR' },
  { name: 'Germany', code: 'DE' },
  { name: 'India', code: 'IN' },
  { name: 'Japan', code: 'JP' },
  { name: 'Spain', code: 'ES' },
  { name: 'United States', code: 'US' },
]);
</script>

<template>
  <main>
    <MultiSelect
      v-model="selectedCountries"
      :options="countries"
      optionLabel="name"
      display="chip"
    >
      <template #value="{ value }">
        My overriden value: {{ value.name }}
      </template>
    </MultiSelect>
  </main>
</template>

<style scoped></style>
```

### Expected behavior

Rendered MultiSelect where selected chips values are "My overridden value: {{selected country name}}"

**Comment by Peanut4287** [comment]
@reijosirila
 The behavior you are seeing in v4 is exactly the same as in v3.
If you want to adjust only the value inside the chip, you need to use the `chip` slot, not the `value` slot.
The `chip` slot only receives a single selected value, but the `value` slot receives the whole array.
But again, this is also true for v3.

TL;DR: use the `chip` slot

**Comment by blouflashdb** [comment]
When upgrading to v4 i had to change the chip slot from:
```
<template #chip="slotProps">
      <i :class="slotProps.value.Icon" class="me-1" aria-hidden="true"></i>{{ slotProps.value.DisplayText }}
</template>
```
to:
```
<template #chip="slotProps">
      <Chip :label="slotProps.value.DisplayText" :icon="slotProps.value.Icon" removable @remove="slotProps.removeCallback" />
</template>
```
in order for it to be displayed like before

**Comment by tugcekucukoglu** [comment]
`value` attribute of `value` slot has the same type as `modelValue`. Since they are arrays, you can use a for loop.
